types/node/index.d.ts

```diff
@@ -824,7 +825,7 @@ declare namespace NodeJS {
         emit(event: "unhandledRejection", reason: any, promise: Promise<any>): boolean;
         emit(event: "warning", warning: Error): boolean;
         emit(event: "message", message: any, sendHandle: any): this;
-        emit(event: Signals): boolean;
+        emit(event: Signals, signal: Signals): boolean;
```

Comment:
I'm a big confused over the usage here. Can you add some tests of how this is intended to be used?

Reply:
@DanielRosenwasser if you look at the type `SignalsListener` you see that it receives the `signal` as first argument (see also https://nodejs.org/dist/latest-v10.x/docs/api/process.html#process_signal_events). To actually emit this it's needed to pass the signal as argument to `emit()` (see https://nodejs.org/dist/latest-v10.x/docs/api/events.html#events_emitter_emit_eventname_args).
User code will usually not emit one of the Signals events as they come from OS but at least for tests it's valid to do this manually.

Reply:
I am a little skeptical that we should add this, especially when it's not totally complete for its arguments (e.g `SIGWINCH` should require another argument as well). But if there's not a catch-all signature, you might as well add them.